lib/websocket.go

```diff
@@ -20,7 +44,16 @@ type WebsocketHandler interface {
 // and serves to maintain the list of connections
 type wsHandler struct {
 	// Collect all websocket connections
-	conns []*websocket.Conn
+	conns         map[string]*wsConn
+	connsLock     sync.Mutex
+	keystore      key.Store
+	subscriptions map[string]string
```

Comment:
an interesting side effect of this design: each profile can have at most one websocket connection, which will cause two browser tabs from the same profile to fail. Let's add a TODO to support multiple connections for the same profile, which will require an array of connection ID strings as the value instead of a single string.

We need it someday, but not to get an MVP out the door